### ISSUE TITLE
Host endpoints created by automatic host endpoints feature have no IPs in iptables mode on EKS with AWS VPC CNI

### ISSUE THREAD

**Issue by chingyi-lin** [projectcalico/calico#4721]
## Expected Behavior

Host endpoints created by automatic host endpoints feature ([doc](https://docs.projectcalico.org/security/kubernetes-nodes#enable-automatic-host-endpoints)) should have node IPs.

## Current Behavior
Host endpoints created by automatic host endpoints feature doesn't have node IPs; for example, it is how it looks like
```
calicoctl get hep -owide

NAME                                   NODE                          INTERFACE   IPS   PROFILES
ip-10-0-25-6.ec2.internal-auto-hep     ip-10-0-25-6.ec2.internal     *                 projectcalico-default-allow
```

## Possible Solution

One workaround is to temporarily enable eBPF mode and then switch back to iptables. Then enable the automatic host endpoints.  

It seems after switching the modes, the `installation.operator.tigera.io` has an extra entry `nodeAddressAutodetectionV4` in the `spec`:
```
  spec:
    calicoNetwork:
      bgp: Disabled
      linuxDataplane: Iptables
      nodeAddressAutodetectionV4:
        canReach: 8.8.8.8
    cni:
      ipam:
        type: AmazonVPC
      type: AmazonVPC
```
Besides, before applying the workaround, `calicoctl get nodes -oyaml` has `spec` as
```
  spec:
    addresses:
    - address: 10.0.35.209
      type: InternalIP
```
but after applying the workaround, the node spec has `CalicoNodeIP`.
```
  spec:
    addresses:
    - address: 10.0.59.61/20
      type: CalicoNodeIP
    - address: 10.0.59.61
      type: InternalIP
```
## Steps to Reproduce (for bugs)

1. Create a eks 1.18 cluster
2. Install calico via operator based installation - Follow the steps in https://docs.projectcalico.org/maintenance/ebpf-and-eks to install Calico with the AWS VPC CNI but do not disable kube-proxy or enable eBPF.
3. Enable automatic host endpoints as documented [here](https://docs.projectcalico.org/security/kubernetes-nodes#enable-automatic-host-endpoints):

```
calicoctl patch kubecontrollersconfiguration default --patch='{"spec": {"controllers": {"node": {"hostEndpoint": {"autoCreate": "Enabled"}}}}}'
```

4. Execute `calicoctl get hep -owide` and observe no IPs are presented.

## Context

We want to use automatic host endpoints to detect node ips and create the endpoints so we can start enforcing policy that involves host networking.
 
## Your Environment

* Calico version: 3.19.1
* Orchestrator version (e.g. kubernetes, mesos, rkt): eks 1.18.16 with AWS VPC CNI 1.7.5
* Operating System and version: Amazon Linux 2, kernel version 4.14.203
* Link to your project (optional): n/a

**Comment by lmm** [comment]
We chatted about this over in Calico Users slack. This is at least a docs bug because I don't think we document that auto host endpoints only work for clusters running Calico CNI with BGP enabled.

We're planning on improving the way the auto host endpoints `expectedIPs` field is populated.

**Comment by chingyi-lin** [comment]
Update: The workaround I discovered are not sufficient for our use case because it only applies to the existing nodes. As our clusters have auto-scaling enabled, when the cluster scales, the new nodes won’t get the IP, and then the host policy will be broken. Wondering what the magic the eBPF mode does behind the scene, maybe we can patch some calico custom resource to achieve the same in iptables mode?

**Comment by caseydavenport** [comment]
The fix here should be very small - we just need to update our controller to look at the full list of IP addresses on the node. Right now, it is only looking at fields that are populated with Calico networking enabled.

**Comment by caseydavenport** [comment]
@coutinhop I believe this will be fixed by the work you're doing right now.

**Comment by coutinhop** [comment]
https://github.com/projectcalico/calico/pull/5405 just merged into master, will go in v3.22
